actionview/lib/action_view/helpers/cache_helper.rb

```diff
@@ -292,21 +292,23 @@ def write_fragment_for(name, options)
         controller.write_fragment(name, fragment, options)
       end
 
-      class CachingRegistry
-        extend ActiveSupport::PerThreadRegistry
+      module CachingRegistry
+        extend self
 
-        attr_accessor :caching
-        alias caching? caching
+        def caching?
+          ActiveSupport::IsolatedExecutionState[:action_view_caching] ||= false
+        end
 
-        def self.track_caching
-          caching_was = self.caching
-          self.caching = true
+        def track_caching
+          caching_was = ActiveSupport::IsolatedExecutionState[:action_view_caching]
+          ActiveSupport::IsolatedExecutionState[:action_view_caching] = true
 
           yield
         ensure
-          self.caching = caching_was
+          ActiveSupport::IsolatedExecutionState[:action_view_caching] = caching_was
         end
       end
+      private_constant :CachingRegistry
```

Comment:
Do we need this `private_constant`? We avoid to use it in Rails since there is no much point of making constants private. It should be `# :nodoc` though.